### ISSUE TITLE
After successfully accepting an invitation to a direct room, the permissions are not updated in a timely manner

### ISSUE THREAD

**Issue by ZhangHanDong** [project-robius/robrix#302]
![image](https://github.com/user-attachments/assets/a88fd9cb-9604-4dc0-aa8f-1e4e6903d60f)

Currently, there are two checks for permissions: one when selecting a room and another when there is a change in permissions within the room. After successfully accepting an invitation to a direct room, the permissions are not updated in a timely manner.

**Comment by ZhangHanDong** [comment]
[Relevant PR #253](https://github.com/project-robius/robrix/pull/253)

**Comment by kevinaboos** [comment]
@aaravlu please add this issue to the list of tasks for you to fix up after #253 was merged.

**Comment by kevinaboos** [comment]
@aaravlu Perhaps we *should* choose to make the default be `true` for this, since if we don't yet know what the user's permissions are in a given room, it's better to show the regular message input box than an incorrect permissions warning that will confuse the user.

**Comment by aaravlu** [comment]
> Perhaps we _should_ choose to make the default be `true` for this

Got all. NP!

**Comment by aaravlu** [comment]
![Screenshot from 2024-12-27 18-27-18](https://github.com/user-attachments/assets/016f04c2-30c9-4cdb-8b04-f02174d0b181)
![Screenshot from 2024-12-27 18-26-12](https://github.com/user-attachments/assets/56eca03e-ff05-47ce-a1b9-560ce9f51500)

IMHO, robrix's invite-to-be-accepted RoomScreen should be aligned with element, i.e. it should not show a prompt, nor an input box. It should show a new interface. How do you like it?

**Comment by kevinaboos** [comment]
> ![Screenshot from 2024-12-27 18-27-18](https://private-user-images.githubusercontent.com/157377732/398917439-016f04c2-30c9-4cdb-8b04-f02174d0b181.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUzMjk5MzQsIm5iZiI6MTczNTMyOTYzNCwicGF0aCI6Ii8xNTczNzc3MzIvMzk4OTE3NDM5LTAxNmYwNGMyLTMwYzktNGNkYi04YjA0LWYwMjE3NGQwYjE4MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIyN1QyMDAwMzRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNTFkNjc0MzAyYjRjMTQ2OGM0NDkyODM3MzFmNzMzOTRlMDY4NGE4NTEzMTJhODc3OTM2MzJmMTYyYWM0ZGYwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.GegjlcsxajjV9oEpKvY2jCpgYApAbHo0c8uWlc7l1OU) ![Screenshot from 2024-12-27 18-26-12](https://private-user-images.githubusercontent.com/157377732/398917448-56eca03e-ff05-47ce-a1b9-560ce9f51500.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUzMjk5MzQsIm5iZiI6MTczNTMyOTYzNCwicGF0aCI6Ii8xNTczNzc3MzIvMzk4OTE3NDQ4LTU2ZWNhMDNlLWZmMDUtNDdjZS1hMWI5LTU2MGNlOWY1MTUwMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIyN1QyMDAwMzRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYmZjZWYxOTZkYzE3ZmNjNmQ2MzM0YzA2N2IwZDFhMDljOGU2ZDgxOGViMjhmM2IzMTAzYjcyZmE5MzNiYTIxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.A_rDmAzvwlIzAe6MWNtdgit6GRchD83i_m269NihI9A)
> 
> IMHO, robrix's invite-to-be-accepted RoomScreen should be aligned with element, i.e. it should not show a prompt, nor an input box. It should show a new interface. How do you like it?

yes, we should show a completely new screen for handling invitations, but that is a separate issue that requires a lot of complexity. Let's save that for later, after our first milestone is complete in early February.

**Comment by kevinaboos** [comment]
For now, let's just keep this issue focused on updating the room power levels quickly and efficiently, just to ensure that we don't incorrectly tell a user that they cannot send messages in a room where they actually can.
